firebase-firestore/src/main/java/com/google/firebase/firestore/ReadOnlyTransaction.java

```diff
@@ -0,0 +1,43 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.firestore;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.google.firebase.Timestamp;
+
+public class ReadOnlyTransaction {
+
+  ReadOnlyTransaction() {
+  }
+
+  @NonNull
+  public DocumentSnapshot get(@NonNull DocumentReference documentRef)
+      throws FirebaseFirestoreException {
+    return get(documentRef, null);
+  }
+
+  @NonNull
+  public DocumentSnapshot get(@NonNull DocumentReference documentRef, @Nullable Timestamp readTime)
```

Comment:
FWIW While we could implement this, this breaks the transaction concept of using the same read time across all reads.

Reply:
Okay, I'll fix this.

Reply:
Thanks. We need to ensure that a transaction always returns a consistent view of data.

Reply:
Done.